TT-1768: First draft of documentation

J-Lambo opened the pull request:
Documented database, matching strategy, and basic run instructions for example local matching service. 

With input from: @SKeerthana @andy-paine

J-Lambo committed:
TT-1768: First draft of documentation  

Documented database, matching strategy, and basic run instructions for example local matching service. 

With input from: @SKeerthana @andy-paine

SKeerthana commented on README.md:
```diff
@@ -1,4 +1,137 @@
+
+The example ignores any attributes where `verified` is set to `false`. This would be a suitable starting point for any service with an existing dataset expected to include most new users.
+
+The example will ignore any verified historical
```
It only ignores any unverified attribute as mentioned above. We are not ignoring the "verified" historical attributes.

J-Lambo committed:
TT-1768: Clarify verified historical attributes

Removed rogue line about not using verified historical attributes as this is false.

J-Lambo commented:
Thanks @SKeerthana. Good spot! I've amended

J-Lambo committed:
TT-1768: Add info on script and matching datasets

Filled some missing gaps on sample matching datasets, script to execute, and additional line on address and surname history.

andy-paine commented on README.md:
```diff
@@ -1,4 +1,145 @@
+# Example local matching service for GOV.UK Verify
 
-This is an example of local matching service that could be used by relaying parties as a reference when building their matching service.
+This is an example local matching service built by the GOV.UK Verify team to demonstrate how a local matching service works and help services building their own. Every government service connecting to Verify must build their own local matching service.
```
I feel like `must build their own` sounds a bit anti-collaborative. Maybe just `must build a local matching service` as in an ideal world we would actually have us and RPs sharing code/helping each other 😄

andy-paine commented on README.md:
```diff
@@ -1,4 +1,145 @@
+
+This example local matching service uses a SQL database. Refer to the [database schema](/docs/schema.png) for details.
+
+All SQL has been written for Postgres and has not been tested with any engine.
```
Should say any other engine - we've tested it in Postgres at least

J-Lambo committed:
TT-1768: Fix typos and wording

Amended based on feedback from @andy-paine
